ifermi/plot.py

```diff
@@ -346,7 +353,10 @@ def _get_plot_data(
 
         isosurfaces = []
         if not hide_surface:
-            isosurfaces = self.fermi_surface.all_vertices_faces(spins=spin)
+            if plot_index is None:
+                isosurfaces = self.fermi_surface.all_vertices_faces(spins=spin)
+            else:
+                isosurfaces = [self.fermi_surface.all_vertices_faces(spins=spin)[int(i)] for i in plot_index]
```

Comment:
I.e., this if statement would just become

```suggestion
            isosurfaces = self.fermi_surface.all_vertices_faces(spins=spin, band_index=plot_index)
```

And the property selection a few lines below would become:

```python
properties = self.fermi_surface.all_properties(
    spins=spin, band_index=plot_index, projection_axis=projection_axis, norm=norm
)
```